roles/adoption_osp_deploy/templates/tripleo-ansible-inventory.yaml.j2

```diff
@@ -0,0 +1,36 @@
+#jinja2: trim_blocks:True, lstrip_blocks:True
+{% for group, role in _adoption_source_scenario.roles_groups_map.items() %}
+{{ role }}:
+  hosts:
+    {% for node in _vm_groups[group] %}
+    {% set node_nets = cifmw_networking_env_definition.instances[node] %}
+    {{ node }}:
+      ansible_host: {{ node_nets.networks.ctlplane.ip_v4 }}
+      canonical_hostname: {{ node }}.{{ _adoption_source_scenario.cloud_domain }}
+      ctlplane_ip: {{ node_nets.networks.ctlplane.ip_v4 }}
+      {% for network_name, net in node_nets.networks.items() %}
+      {% if 'vlan_id' in net %}
+        {% set net_name = ['storage_mgmt'] if network_name == 'storagemgmt' else [network_name] %}
+        {% set net_name = ['internal_api'] if network_name == 'internalapi' else [network_name] %}
+      {{ net_name[0] }}_ip: {{ net.ip_v4 }}
+      {% endif %}
+      {% endfor %}
+    {% endfor %}
+  vars:
+    ansible_ssh_common_args: -o StrictHostKeyChecking=no
+    ansible_ssh_user: zuul
+{% endfor %}
+Undercloud:
+  hosts:
+    undercloud: {}
+  vars:
+    ansible_connection: local
+    ansible_host: localhost
+allovercloud:
+  children:
+    {% for _, role in _adoption_source_scenario.roles_groups_map.items() %}
+    {{ role }}: {}
+    {% endfor %}
+computes:
+  children:
+    {{ _adoption_source_scenario.roles_groups_map['osp-computes'] }}: {}
```

Comment:
I'm wondering if this inventory can be used in place of [1] when we trigger the ceph migration (after the adoption) and a playbook that is meant to prepare the TripleO environment [2].
This would save us time to build a "ceph compatibility" layer when it's about executing those playbooks. Do you think we can extend this to have a Ceph section? Maybe worth discussing this in a follow up.

[1] https://review.rdoproject.org/r/c/rdo-jobs/+/53695/59/playbooks/data_plane_adoption/templates/ceph_inventory.j2
[2] https://github.com/openstack-k8s-operators/data-plane-adoption/pull/637

Reply:
@fmount I can try adding a new section matching what you have in `ceph_inventory` once I finish the current testing run, and check that it doesn't interfere with what I currently have. If that works we can get it in on this first version, if we need further changes we can leave it for a follow-up

Reply:
ack and thank you!

Reply:
@fmount I think I was able to make the change, and the overcloud deployed fine, here is the resulting inventory, let me know if that looks ok for your usecase:

```
ComputeHCI:
  hosts:
    osp-compute-l6eicnz2-0:
      ansible_host: 192.168.122.106
      canonical_hostname: osp-compute-l6eicnz2-0.localdomain
      ctlplane_ip: 192.168.122.106
      external_ip: 172.21.0.106
      internal_api_ip: 172.17.0.106
      storage_ip: 172.18.0.106
      storagemgmt_ip: 172.20.0.106
      tenant_ip: 172.19.0.106
    osp-compute-l6eicnz2-1:
      ansible_host: 192.168.122.107
      canonical_hostname: osp-compute-l6eicnz2-1.localdomain
      ctlplane_ip: 192.168.122.107
      external_ip: 172.21.0.107
      internal_api_ip: 172.17.0.107
      storage_ip: 172.18.0.107
      storagemgmt_ip: 172.20.0.107
      tenant_ip: 172.19.0.107
    osp-compute-l6eicnz2-2:
      ansible_host: 192.168.122.108
      canonical_hostname: osp-compute-l6eicnz2-2.localdomain
      ctlplane_ip: 192.168.122.108
      external_ip: 172.21.0.108
      internal_api_ip: 172.17.0.108
      storage_ip: 172.18.0.108
      storagemgmt_ip: 172.20.0.108
      tenant_ip: 172.19.0.108
  vars:
    ansible_ssh_common_args: -o StrictHostKeyChecking=no
    ansible_ssh_user: zuul
Controller:
  hosts:
    osp-controller-l6eicnz2-0:
      ansible_host: 192.168.122.103
      canonical_hostname: osp-controller-l6eicnz2-0.localdomain
      ctlplane_ip: 192.168.122.103
      external_ip: 172.21.0.103
      internal_api_ip: 172.17.0.103
      storage_ip: 172.18.0.103
      storagemgmt_ip: 172.20.0.103
      tenant_ip: 172.19.0.103
    osp-controller-l6eicnz2-1:
      ansible_host: 192.168.122.104
      canonical_hostname: osp-controller-l6eicnz2-1.localdomain
      ctlplane_ip: 192.168.122.104
      external_ip: 172.21.0.104
      internal_api_ip: 172.17.0.104
      storage_ip: 172.18.0.104
      storagemgmt_ip: 172.20.0.104
      tenant_ip: 172.19.0.104
    osp-controller-l6eicnz2-2:
      ansible_host: 192.168.122.105
      canonical_hostname: osp-controller-l6eicnz2-2.localdomain
      ctlplane_ip: 192.168.122.105
      external_ip: 172.21.0.105
      internal_api_ip: 172.17.0.105
      storage_ip: 172.18.0.105
      storagemgmt_ip: 172.20.0.105
      tenant_ip: 172.19.0.105
  vars:
    ansible_ssh_common_args: -o StrictHostKeyChecking=no
    ansible_ssh_user: zuul
Undercloud:
  hosts:
    undercloud: {}
  vars:
    ansible_connection: local
    ansible_host: localhost
allovercloud:
  children:
    ComputeHCI: {}
    Controller: {}
computes:
  children:
    ComputeHCI: {}
overcloud:
  hosts:
    osp-controller-l6eicnz2-0:
      ansible_user: tripleo-admin
      ansible_host: 192.168.122.103
    osp-controller-l6eicnz2-1:
      ansible_user: tripleo-admin
      ansible_host: 192.168.122.104
    osp-controller-l6eicnz2-2:
      ansible_user: tripleo-admin
      ansible_host: 192.168.122.105
    osp-compute-l6eicnz2-0:
      ansible_user: tripleo-admin
      ansible_host: 192.168.122.106
    osp-compute-l6eicnz2-1:
      ansible_user: tripleo-admin
      ansible_host: 192.168.122.107
    osp-compute-l6eicnz2-2:
      ansible_user: tripleo-admin
      ansible_host: 192.168.122.108
```

Reply:
awesome thank you! By doing this after the overcloud is deployed we can still point to the ceph bits w/o major changes!